tests/testthat/test-indentation_linter.R

```diff
@@ -0,0 +1,445 @@
+test_that("indentation linter flags unindented expressions", {
```

Comment:
another test I'd like to see, make sure that multiple consecutive "indenting" operators only outdent by one level, e.g.

```r
local({
  some_code
})
```

is OK, and the logic isn't thinking the indentation should be

```r
local({
    some_code
})
```

(I think this is already handled / would have been picked up from your spot checks on other packages, but it'll be good to keep as a regression test if we ever refactor)